File: 2609.go
func findTheLongestBalancedSubstring(s string) int {
	var count0, count1, answer int
	var lastChar string
	for i := 0; i < len(s); i++ {
		if s[i:i+1] == "0" {
			if s[i:i+1] != lastChar {
				answer = max(answer, min(count0, count1))
				count0 = 0
				count1 = 0
			}
			count0++
		} else {
			if s[i:i+1] != lastChar {
				count1 = 0
			}
			count1++
		}
		lastChar = s[i : i+1]
	}

	answer = max(answer, min(count0, count1))
	return answer * 2
}

File: 2609.php
class Solution {

    /**
    * @param String $s
    * @return Integer
    */
    function findTheLongestBalancedSubstring($s) {
        $lastChar = "";
        $ans = 0;
        $m = 0;
        $n = 0;
        for($i = 0; $i < strlen($s); $i++) {
            if ($lastChar == "1" && $s[$i] == "0") {
                $ans =max($ans, $n-abs($m));
                $m = 0;
                $n =0;
            }
            if ($s[$i] == "0") {
                $m++;
            } else {
                $m--;
            }
            $n++;
            $lastChar = $s[$i];
        } 
        $ans =max($ans, $n-abs($m));
        return $ans;
    }
}